### ISSUE TITLE
Feature: Alternate goal unlock conditions

### ISSUE THREAD

**Issue by jeffreybardon** [Thedragon005/Archipelago-Ape-Escape#11]
### What feature would you like to see?

Currently, "go mode" is just having the required gadgets and access to the level with the boss. This works pretty well for Specter 1 as a goal, but not so well for Specter 2. Peak Point Matrix unlocks with 6 world keys, while the vanilla condition is having caught all monkeys. Having played a few seeds as though the condition was all monkeys, the endgame gets a bit stale, and full completion isn't the most interesting condition for a randomizer.

There are 3 main options to support here:
1. No additional conditions (just the world keys)
2. X level completions (0-19)
3. X "triforce pieces" (Specter Coins) (both amount available and required can be set)

Locking Specter 1 in a similar way would be a nice stretch goal, as this would require finding new memory addresses to prevent the fight from starting until the condition is met.

**Comment by Thedragon005** [comment]
In discord we talked about the option to get "Monkey" as items for the unlock, a little like Power Stars from SM64 . After #16 is implemented we could custom gate CC and TVT bosses to X of the "Monkey" multiworld item, also like SM64

**Comment by Thedragon005** [comment]
User Cynichill proposed an idea to select at random monkeys that would be unlock conditions :[Discord message](https://discord.com/channels/731205301247803413/1168645708103028856/1293634531613606058)
Maybe random int between 1 and 204 to select monkeys until you have X percentage of monkeys,rounded down. Store in slot_data for easy tracking/check which monkeys you need?

**Comment by jeffreybardon** [comment]
Quick skeleton of how the options could be laid out in a way that's both flexible - to let people play how they want - and simple to convey:

bosscondition: The client-enforced condition to allow entering/starting the goal boss. World Keys are always required.
    vanilla: The vanilla condition. Specter 1 is just getting there, Specter 2 is all monkeys.
    levels: Catching all monkeys in some amount of levels. (this may not be implemented in the first pass)
    pieces: Collecting some amount of "triforce pieces" (name TBD but probably Specter Coins) throughout the multiworld
    
requiredlevels:
    can be a value from 0 to 19: if the boss condition is levels, the number of completed levels required to start the boss fight
    
requiredpieces: The number of required pieces to start the boss fight
totalpieces: The total number of pieces available
piecelocations: The types of locations that are valid for "triforce pieces" to be placed in
    monkeys: Specifically local monkey locations
    ownworld: anywhere in your world
    anywhere: anywhere in the multiworld

Additionally, a new "token" choice will be added to the existing "goal" option, where the multiworld victory condition is collecting enough tokens. Both bosses will be available via their vanilla conditions.

**Comment by Thedragon005** [comment]
Note on this : Since a little progress has been made for Room ER we could potentially disable the transition completly (Even accounting room ER) to prevent player from entering S1

**Comment by jeffreybardon** [comment]
I think that due to that same ER consideration, the method of blocking the fight should be contained within the room, as the entrance to this room could be anywhere, potentially.
